README
NAME
    DBIx::Class::InflateColumn::Serializer::Sereal - Sereal based
    Serialization for DBIx Class Columns

VERSION
    version 0.001002

SYNOPSIS
    Standard DBIx::Class definition:

      package Some::Result::Item;
      use parent 'DBIx::Class::Core';

      # Add Inflate::Column::Serializer to component loading.
      __PACKAGE__->load_components( 'InflateColumn::Serializer', 'Core' );
      __PACKAGE__->table('item');
      __PACKAGE__->add_column( itemid => { data_type => 'integer' }, );
      __PACKAGE__->set_primary_key('itemid');
      __PACKAGE__->add_column(
        data => {
          data_type        => 'text',
          size             => 1024,
          serializer_class => 'Sereal', # This line tells InflateColumn::Serializer what class to use.
        }
      );
      __PACKAGE__->source_name('Item');

METHODS
  get_freezer
    This is an implementation detail for the "InflateColumn::Serializer"
    module.

       my $freezer = ::Sereal->get_freezer( $column, $info, $column_args );
       my $string = $freezer->( $object );
       # $data isa string

  get_unfreezer
    This is an implementation detail for the "InflateColumn::Serializer"
    module.

        my $unfreezer = ::Sereal>get_unfreezer( $column, $info, $args );
        my $object = $unfreezer->( $string );

AUTHOR
    Kent Fredric <kentnl@cpan.org>

COPYRIGHT AND LICENSE
    This software is copyright (c) 2014 by Kent Fredric
    <kentfredric@gmail.com>.

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
Release history for DBIx-Class-InflateColumn-Serializer-Sereal

0.001003 2017-03-07T20:11:30Z 346f2a8

0.001002 2017-03-07T20:08:03Z a020378
 - Bugfix: Avoid test failures with -Ddefault_inc_excludes_dot

 [Dependencies::Stats]
 - Dependencies changed since 0.001001, see misc/*.deps* for details
 - configure: +1 (recommends: ↑1)
 - develop: +11 ↑3 -3 (recommends: +1, suggests: ↑2)
 - test: +1 ↓1 (recommends: +1 ↑1 ↓1)

0.001001 2014-10-20T03:29:42Z be70c3d
 [Bugfix]
 - State a minimum version of Sereal components for _with_object function.

 [Dependencies::Stats]
 - Dependencies changed since 0.001000, see misc/*.deps* for details
 - develop: (suggests: ↑1)
 - runtime: ↑2
 - test: (recommends: ↑1)

0.001000 2014-10-18T21:52:31Z f738bbd
 - First version.


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.009.
use strict;
use warnings;

use 5.006;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Sereal based Serialization for DBIx Class Columns",
  "AUTHOR" => "Kent Fredric <kentnl\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "DBIx-Class-InflateColumn-Serializer-Sereal",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.006",
  "NAME" => "DBIx::Class::InflateColumn::Serializer::Sereal",
  "PREREQ_PM" => {
    "Carp" => 0,
    "Sereal::Decoder" => "2.070000",
    "Sereal::Encoder" => "2.070000",
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "DBD::SQLite" => 0,
    "DBI" => 0,
    "DBIx::Class::Core" => 0,
    "DBIx::Class::InflateColumn::Serializer" => 0,
    "DBIx::Class::Schema" => 0,
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "Path::Tiny" => 0,
    "Test::Fatal" => 0,
    "Test::More" => 0,
    "parent" => 0
  },
  "VERSION" => "0.001003",
  "test" => {
    "TESTS" => "t/*.t t/00-compile/*.t"
  }
);


my %FallbackPrereqs = (
  "Carp" => 0,
  "DBD::SQLite" => 0,
  "DBI" => 0,
  "DBIx::Class::Core" => 0,
  "DBIx::Class::InflateColumn::Serializer" => 0,
  "DBIx::Class::Schema" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "Path::Tiny" => 0,
  "Sereal::Decoder" => "2.070000",
  "Sereal::Encoder" => "2.070000",
  "Test::Fatal" => 0,
  "Test::More" => 0,
  "parent" => 0,
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);


maint/perlcritic.rc.gen.pl
#!/usr/bin/env perl
## no critic (Modules::RequireVersionVar)

# ABSTRACT: Write an INI file from a bundle

use 5.008;    # utf8
use strict;
use warnings;
use utf8;

our $VERSION = 0.001;

use Carp qw( croak carp );
use Perl::Critic::ProfileCompiler::Util qw( create_bundle );
use Path::Tiny qw(path);

## no critic (ErrorHandling::RequireUseOfExceptions)
my $bundle = create_bundle('Example::Author::KENTNL');
$bundle->configure;

my @stopwords = (qw());
for my $wordlist (@stopwords) {
  $bundle->add_or_append_policy_field( 'Documentation::PodSpelling' => ( 'stop_words' => $wordlist ) );
}

#$bundle->add_or_append_policy_field(
#  'Subroutines::ProhibitCallsToUndeclaredSubs' => ( 'exempt_subs' => 'String::Formatter::str_rf' ), );

$bundle->remove_policy('ErrorHandling::RequireUseOfExceptions');
$bundle->remove_policy('CodeLayout::RequireUseUTF8');
#$bundle->remove_policy('ErrorHandling::RequireCarping');
#$bundle->remove_policy('NamingConventions::Capitalization');

my $inf = $bundle->actionlist->get_inflated;

my $config = $inf->apply_config;

{
  my $rcfile = path('./perlcritic.rc')->openw_utf8;
  $rcfile->print( $config->as_ini, "\n" );
  close $rcfile or croak 'Something fubared closing perlcritic.rc';
}
my $deps = $inf->own_deps;
{
  my $target = path('./misc');
  $target->mkpath if not $target->is_dir;

  my $depsfile = $target->child('perlcritic.deps')->openw_utf8;
  for my $key ( sort keys %{$deps} ) {
    $depsfile->printf( "%s~%s\n", $key, $deps->{$key} );
    *STDERR->printf( "%s => %s\n", $key, $deps->{$key} );
  }
  close $depsfile or carp 'Something fubared closing perlcritic.deps';
}



lib/DBIx/Class/InflateColumn/Serializer/Sereal.pm
use 5.006;    # our
use strict;
use warnings;

package DBIx::Class::InflateColumn::Serializer::Sereal;

our $VERSION = '0.001003';

# ABSTRACT: Sereal based Serialization for DBIx Class Columns

our $AUTHORITY = 'cpan:KENTNL'; # AUTHORITY

use Sereal::Encoder 2.070000 qw( sereal_encode_with_object );
use Sereal::Decoder 2.070000 qw( sereal_decode_with_object );
use Carp qw( croak );

sub get_freezer {
  my ( undef, undef, $col_info, undef ) = @_;
  my $encoder = Sereal::Encoder->new();
  if ( defined $col_info->{'size'} ) {
    return sub {
      my $v = sereal_encode_with_object( $encoder, $_[0] );
      croak('Value Serialization is too big')
        if length($v) > $col_info->{'size'};
      return $v;
    };
  }
  return sub {
    return sereal_encode_with_object( $encoder, $_[0] );
  };
}

sub get_unfreezer {
  my $decoder = Sereal::Decoder->new();
  return sub {
    return sereal_decode_with_object( $decoder, $_[0] );
  };
}

1;

__END__

=pod

=encoding UTF-8

=head1 NAME

DBIx::Class::InflateColumn::Serializer::Sereal - Sereal based Serialization for DBIx Class Columns

=head1 VERSION

version 0.001003

=head1 SYNOPSIS

Standard DBIx::Class definition:

  package Some::Result::Item;
  use parent 'DBIx::Class::Core';

  # Add Inflate::Column::Serializer to component loading.
  __PACKAGE__->load_components( 'InflateColumn::Serializer', 'Core' );
  __PACKAGE__->table('item');
  __PACKAGE__->add_column( itemid => { data_type => 'integer' }, );
  __PACKAGE__->set_primary_key('itemid');
  __PACKAGE__->add_column(
    data => {
      data_type        => 'text',
      size             => 1024,
      serializer_class => 'Sereal', # This line tells InflateColumn::Serializer what class to use.
    }
  );
  __PACKAGE__->source_name('Item');

=head1 METHODS

=head2 get_freezer

This is an implementation detail for the C<InflateColumn::Serializer> module.

   my $freezer = ::Sereal->get_freezer( $column, $info, $column_args );
   my $string = $freezer->( $object );
   # $data isa string

=head2 get_unfreezer

This is an implementation detail for the C<InflateColumn::Serializer> module.

    my $unfreezer = ::Sereal>get_unfreezer( $column, $info, $args );
    my $object = $unfreezer->( $string );

=head1 AUTHOR

Kent Fredric <kentnl@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2017 by Kent Fredric <kentfredric@gmail.com>.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut
